src/library_manager/llext_manager.c

```diff
@@ -268,56 +270,63 @@ static int llext_manager_link(const char *name,
 			.section_detached = llext_manager_section_detached,
+			.keep_section_info = true,
 		};
 
 		ret = llext_load(ldr, name, llext, &ldr_parm);
 		if (ret)
 			return ret;
 	}
 
-	mctx->segment[LIB_MANAGER_TEXT].addr = ldr->sects[LLEXT_MEM_TEXT].sh_addr;
-	mctx->segment[LIB_MANAGER_TEXT].size = ldr->sects[LLEXT_MEM_TEXT].sh_size;
+	/* All code sections */
+	llext_get_region_info(ldr, *llext, LLEXT_MEM_TEXT, &hdr, NULL, NULL);
+	mctx->segment[LIB_MANAGER_TEXT].addr = hdr->sh_addr;
+	mctx->segment[LIB_MANAGER_TEXT].size = hdr->sh_size;
 
 	tr_dbg(&lib_manager_tr, ".text: start: %#lx size %#x",
 	       mctx->segment[LIB_MANAGER_TEXT].addr,
 	       mctx->segment[LIB_MANAGER_TEXT].size);
 
 	/* All read-only data sections */
-	mctx->segment[LIB_MANAGER_RODATA].addr =
-		ldr->sects[LLEXT_MEM_RODATA].sh_addr;
-	mctx->segment[LIB_MANAGER_RODATA].size = ldr->sects[LLEXT_MEM_RODATA].sh_size;
+	llext_get_region_info(ldr, *llext, LLEXT_MEM_RODATA, &hdr, NULL, NULL);
+	mctx->segment[LIB_MANAGER_RODATA].addr = hdr->sh_addr;
+	mctx->segment[LIB_MANAGER_RODATA].size = hdr->sh_size;
 
 	tr_dbg(&lib_manager_tr, ".rodata: start: %#lx size %#x",
 	       mctx->segment[LIB_MANAGER_RODATA].addr,
 	       mctx->segment[LIB_MANAGER_RODATA].size);
 
 	/* All writable data sections */
-	mctx->segment[LIB_MANAGER_DATA].addr =
-		ldr->sects[LLEXT_MEM_DATA].sh_addr;
-	mctx->segment[LIB_MANAGER_DATA].size = ldr->sects[LLEXT_MEM_DATA].sh_size;
+	llext_get_region_info(ldr, *llext, LLEXT_MEM_DATA, &hdr, NULL, NULL);
+	mctx->segment[LIB_MANAGER_DATA].addr = hdr->sh_addr;
+	mctx->segment[LIB_MANAGER_DATA].size = hdr->sh_size;
 
 	tr_dbg(&lib_manager_tr, ".data: start: %#lx size %#x",
 	       mctx->segment[LIB_MANAGER_DATA].addr,
 	       mctx->segment[LIB_MANAGER_DATA].size);
 
-	mctx->segment[LIB_MANAGER_BSS].addr = ldr->sects[LLEXT_MEM_BSS].sh_addr;
-	mctx->segment[LIB_MANAGER_BSS].size = ldr->sects[LLEXT_MEM_BSS].sh_size;
+	/* Writable uninitialized data section */
+	llext_get_region_info(ldr, *llext, LLEXT_MEM_BSS, &hdr, NULL, NULL);
+	mctx->segment[LIB_MANAGER_BSS].addr = hdr->sh_addr;
+	mctx->segment[LIB_MANAGER_BSS].size = hdr->sh_size;
 
 	tr_dbg(&lib_manager_tr, ".bss: start: %#lx size %#x",
 	       mctx->segment[LIB_MANAGER_BSS].addr,
 	       mctx->segment[LIB_MANAGER_BSS].size);
 
 	*buildinfo = NULL;
-	ssize_t binfo_o = llext_find_section(ldr, ".mod_buildinfo");
-
-	if (binfo_o >= 0)
-		*buildinfo = llext_peek(ldr, binfo_o);
+	ret = llext_section_shndx(ldr, *llext, ".mod_buildinfo");
+	if (ret >= 0) {
+		llext_get_section_info(ldr, *llext, ret, &hdr, NULL, NULL);
+		*buildinfo = llext_peek(ldr, hdr->sh_offset);
+	}
 
 	*mod_manifest = NULL;
-	ssize_t mod_o = llext_find_section(ldr, ".module");
-
-	if (mod_o >= 0)
-		*mod_manifest = llext_peek(ldr, mod_o);
+	ret = llext_section_shndx(ldr, *llext, ".module");
+	if (ret >= 0) {
+		llext_get_section_info(ldr, *llext, ret, &hdr, NULL, NULL);
+		*mod_manifest = llext_peek(ldr, hdr->sh_offset);
+	}
```

Comment:
why is this now better than a single call to llext_find_section(llext_find_section()`?